python/pyarrow/_flight.pyx

```diff
@@ -71,6 +74,18 @@ cdef int check_flight_status(const CStatus& status) nogil except -1:
     return check_status(status)
 
 
+def _munge_grpc_python_error(message):
+    import re
+    pat = re.compile(r'Flight RPC failed with message: (.*). Detail: '
```

Comment:
If we're going to compile the regex, might as well do it only once.

Reply:
no prob